### DISCUSSION TITLE
Declare the return type here

### DISCUSSION THREAD

**Discussion by azizoid** [remix-run/react-router#9474] [Proposals]
we could declare the return type in the Hook call, with 
` const data = useLoaderData<WierdDataProps>()`

`export declare function useLoaderData<T>(): T;`

https://github.com/remix-run/react-router/blob/892238bee052177e78012167c088290b06400206/packages/react-router/lib/hooks.tsx#L749

**Comment by cmmartti** [comment]
Just typecast it with `as`. It's functionally equivalent.

```ts
const data = useLoaderData() as WeirdDataProps;
```

**Reply by azizoid** [reply]
that is fast and temporary solution. Not the correct way.
